xml_converter/test_cases/test_expected_outputs.json

```diff
@@ -0,0 +1,38 @@
+{
+    "can_fade": {
+        "path": "../test_cases/can_fade_tests",
+        "tests": {
```

Comment:
From experience in situations like this, using a dict with a list of items in the format `{myname: {mydata}}` always ends up being more pain then `[{name: myname, ...mydata...}]` in the long term. You dont need to be able to lookup these tests by name so having it as a key it not beneficial and you do need to iterate over the data and a dict is unsorted. (Technically python does guarentee ordering of dicts be the same as the input order but this is not true in all languages (like cpp which will sort them alphabetically, or go which goes out of its way to sort them randomly)). You might also want to do this for the top level testgroups too.